OrbitVulkanLayerClientGgp/LayerData.cpp

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "LayerData.h"
+
+#include <fcntl.h>
+#include <unistd.h>
+
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "OrbitBase/Logging.h"
+#include "google/protobuf/io/zero_copy_stream_impl.h"
+#include "google/protobuf/text_format.h"
+#include "layer_config.pb.h"
+
+using orbit_vulkan_capture_protos::LayerConfig;
+
+namespace {
+constexpr char const* kConfigFileName = "/mnt/developer/orbit_capture_vulkan_layer_config.pb.txt";
+constexpr char const* kOrbitCaptureService = "/mnt/developer/OrbitCaptureGgpService";
+constexpr char const* kLogDirectory = "/var/game/";
+constexpr double kFrameTimeThresholdMillisecondsDefault = 1000.0 / 60.0;
+constexpr uint32_t kCaptureLengthSecondsDefault = 10;
+}  // namespace
+
+void LayerData::Init() {
+  // Load data from config file
+  LOG("Loading  vulkan layer config file from %s", std::string(kConfigFileName));
+
+  // Config is a proto text file
+  int config_file_descriptor = open(kConfigFileName, O_RDONLY);
+  if (config_file_descriptor < 0) {
+    ERROR("Not being able to open file. Default values will be used");
+    return;
+  }
+
+  google::protobuf::io::FileInputStream config_file_stream(config_file_descriptor);
+  if (!google::protobuf::TextFormat::Parse(&config_file_stream, &layer_config_)) {
+    ERROR("Parsing vulkan layer config file. Default values will be used");
```

Comment:
Would it be an option to actually abort at this point? If you can't read the config file, then there is an issue that the user did not intend that way (maybe the config file is malformed and needs to be fixed). Even though there is an error message in the log, continuing here helps hiding the fact that something is wrong.

Reply:
Mmm I am not sure if it is possible to force the abort of the game from the vulkan layer; let me try and let you know.

I decided to print the error and continue with default values because I at the time I considered it would be too extreme to abort the execution of the game because of that buy maybe it would actually make sense, as I guess they would run the layer in the case they actually want to use it. Will check if it is possible to do this.

Reply:
If I use FATAL here it prints the error and the vulkan execution finished and the game does not load, however it does not finish either, it just hangs. Would this behavior be what we want?

Reply:
@ronaldfw What should I do about this?

Reply:
Discussed offline: fine to just print the error.